lib/QubitInformationObjectXmlCache.class.php

```diff
@@ -84,27 +84,43 @@ public function export($resource)
   public function exportAll($options = array())
   {
     $skip = isset($options['skip']) ? $options['skip'] : 0;
-
-    // Get not-root and published information objects
-    $criteria = new Criteria;
-    $criteria->add(QubitInformationObject::ID, QubitInformationObject::ROOT_ID, Criteria::NOT_EQUAL);
-    $criteria = QubitAcl::addFilterDraftsCriteria($criteria);
-    $criteria->addAscendingOrderByColumn(QubitInformationObject::ID); // sort so optional skip will be consistent
-    $criteria->setOffset($skip);
-
-    $results = QubitInformationObject::get($criteria);
+    $limit = isset($options['limit']) ? $options['limit'] : 0;
+
+    // Get not-root and published information objects (sorted by ID so optional
+    // skip/limit will be consistent)
+    $sql = "SELECT i.id FROM ". QubitInformationObject::TABLE_NAME ." i \r
+      INNER JOIN status s ON i.id=s.object_id \r
+      WHERE i.id != :root_id \r
+      AND s.status_id=:status_id AND s.type_id=:type_id \r
+      ORDER BY i.id";
+ 
+    $params = array(
+      ':status_id' => QubitTerm::PUBLICATION_STATUS_PUBLISHED_ID,
+      ':type_id' => QubitTerm::STATUS_TYPE_PUBLICATION_ID,
+      ':root_id' => QubitInformationObject::ROOT_ID,
+    );
 
     $exporting = 0;
 
+    // Apply skip/limit and fetch results
+    Propel::getDB()->applyLimit($sql, $skip, $limit);
```

Comment:
I didn't know about this ;)

Reply:
Yeah, it's new to me too... kind of handy.